src/Dscp/Crypto/Encrypt.hs

```diff
@@ -220,31 +223,41 @@ prepareAEAD pp iv =
        aeadInit aeadMode cipher iv
 
 -- | Encrypt given 'ByteArray' with AES.
--- TODO: use Secure Random from OpenSSL for IV calculation instead.
-encrypt :: ByteArray ba => PassPhrase -> ba -> Encrypted ba
+encrypt :: ByteArrayAccess ba => PassPhrase -> ba -> Encrypted ba
 encrypt pp plaintext =
     let eIV = fromMaybe (error "encrypt: impossible: random IV with invalid size") .
-              makeIV @ByteString . unsafePerformIO $
+              makeIV @ByteString . unsafePerformIO . runSecureRandom $
               getRandomBytes cipherBlkSize
         aead = prepareAEAD pp eIV
+        plaintextBS = BA.convert plaintext
         (eAuthTag, eCiphertext) =
-            aeadSimpleEncrypt aead authHeader plaintext authTagLength
+            aeadSimpleEncrypt aead authHeader plaintextBS authTagLength
```

Comment:
`aeadSimpleEncrypt` returns exactly the same thing as which it accepts :unamused: